cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java

```diff
@@ -75,6 +76,8 @@
     protected static final String ENV_VAR_KAFKA_CONNECT_TLS_AUTH_KEY = "KAFKA_CONNECT_TLS_AUTH_KEY";
     protected static final String ENV_VAR_KAFKA_CONNECT_SASL_PASSWORD_FILE = "KAFKA_CONNECT_SASL_PASSWORD_FILE";
     protected static final String ENV_VAR_KAFKA_CONNECT_SASL_USERNAME = "KAFKA_CONNECT_SASL_USERNAME";
+    protected static final String ENV_VAR_KAFKA_CONNECT_SASL_PASSWORD = "KAFKA_CONNECT_SASL_PASSWORD";
+    protected static final String ENV_VAR_KAFKA_CONNECT_SASL_PLAIN_PASSWORD_FILE = "KAFKA_CONNECT_SASL_PLAIN_PASSWORD_FILE";
```

Comment:
Do we really need to have a new variable for this? Would it be better to reuse the variable `ENV_VAR_KAFKA_CONNECT_SASL_PASSWORD_FILE` which we already have and instead pass the SASL mechanism to the image? That would make it much easier to add in the future support for new SASL authentication mechanisms based on username / password without having a different password field for each of them.

Reply:
`KAFKA_CONNECT_SASL_PASSWORD` can be removed.

`KAFKA_CONNECT_SASL_PLAIN_PASSWORD_FILE` is to differentiate from `KAFKA_CONNECT_SASL_PASSWORD_FILE` for `SCRAM-SHA-512` in docker-images/kafka-connect/scripts/kafka_connect_config_generator.sh 

```bash
if [ -n "$KAFKA_CONNECT_SASL_USERNAME" ] && [ -n "$KAFKA_CONNECT_SASL_PASSWORD_FILE" ]; then
...
SASL_MECHANISM="SCRAM-SHA-512"
```

I didn't have better idea to make difference for those two mechanisms in kafka_connect_config_generator.sh without making bigger changes. Please advise.

Reply:
I personally would add a new environment variable ... something like `KAFKA_CONNECT_SASL_MECHANISM` which would contain either `plain` or `scram-sha-512`. You could set it depending on the authentication type. And you can use it in `kafka_connect_config_generator.sh` to configure either PLAIN or SCRAM-SHA-512. That will let you use `KAFKA_CONNECT_SASL_PASSWORD_FILE` for the password with both mechanisms.

Reply:
updated in 673284a please help review